lib/src/di/injector.dart

```diff
@@ -1,23 +1,37 @@
 import 'package:allthenews/src/data/appinfo/app_info_local_repository.dart';
-import 'package:allthenews/src/data/communication/api_key_local_repository.dart';
+import 'package:allthenews/src/data/communication/api/api_exception_mapper.dart';
+import 'package:allthenews/src/data/communication/api/api_key_local_repository.dart';
+import 'package:allthenews/src/data/communication/api/ny_times_rest_repository.dart';
 import 'package:allthenews/src/data/persistence/shared_preferences_persistence_repository.dart';
 import 'package:allthenews/src/data/settings/settings_local_repository.dart';
 import 'package:allthenews/src/domain/appinfo/app_info_repository.dart';
 import 'package:allthenews/src/domain/authorization/api_key_repository.dart';
 import 'package:allthenews/src/domain/common/persistence/persistence_repository.dart';
+import 'package:allthenews/src/domain/communication/exception_mapper.dart';
+import 'package:allthenews/src/domain/nytimes/ny_times_repository.dart';
 import 'package:allthenews/src/domain/settings/settings_repository.dart';
 import 'package:allthenews/src/ui/common/theme/theme_notifier.dart';
 import 'package:allthenews/src/ui/pages/settings/settings_notifier.dart';
 import 'package:get_it/get_it.dart';
 
+final locator = GetIt.instance;
+
 void injectDependencies() {
-  final locator = GetIt.instance;
-  locator.registerSingleton<ApiKeyRepository>(ApiKeyLocalRepository());
   locator.registerSingleton<AppInfoRepository>(AppInfoLocalRepository());
   locator.registerSingleton<PersistenceRepository>(SharedPreferencesPersistenceRepository());
-  locator.registerSingleton<SettingsRepository>(SettingsLocalRepository(locator<PersistenceRepository>()));
+  locator.registerSingleton<SettingsRepository>(
+      SettingsLocalRepository(locator<PersistenceRepository>()));
   locator.registerFactory(() => ThemeNotifier(locator<SettingsRepository>()));
-  locator.registerFactory(() => SettingsNotifier(locator<SettingsRepository>(), locator<AppInfoRepository>()));
+  locator.registerFactory(
+      () => SettingsNotifier(locator<SettingsRepository>(), locator<AppInfoRepository>()));
+  _injectApiDependencies();
+}
+
+void _injectApiDependencies() {
+  locator.registerSingleton<ApiKeyRepository>(ApiKeyLocalRepository());
+  locator.registerSingleton<NYTimesRepository>(NYTimesRESTRepository());
+  locator.registerSingleton<ExceptionMapper>(ApiExceptionMapper());
 }
 
-T inject<T>([String name]) => GetIt.instance.get<T>(instanceName: name);
+T inject<T>({String name, dynamic param1}) =>
```

Comment:
jak nie ma `param2` to `param` zamiast `param1` mógłby być :P

Reply:
to pozostałość po tworzeniu fabryki do tworzenia obiektów z parametrami, ale tak, zmienię nazwę